core/src/main/java/com/linecorp/armeria/server/HttpServerUpgradeHandler.java

```diff
@@ -322,4 +326,11 @@ private static FullHttpResponse newUpgradeResponse() {
         res.headers().add(HttpHeaderNames.UPGRADE, Http2CodecUtil.HTTP_UPGRADE_PROTOCOL_NAME);
         return res;
     }
+
+    /**
+     * Creates the 400 Bad Request response message.
+     */
+    private static FullHttpResponse newBadRequestResponse() {
+        return new DefaultFullHttpResponse(HTTP_1_1, BAD_REQUEST, Unpooled.EMPTY_BUFFER);
+    }
```

Comment:
Should we return a 400 response with a detail reason phrase?
```suggestion
    private static FullHttpResponse newInvalidSettingsHeadersResponse() {
        return new DefaultFullHttpResponse(HTTP_1_1, BAD_REQUEST,
                                           Unpooled.copiedBuffer("Invalid HTTP2-Settings header",
                                                                 StandardCharsets.UTF_8));
    }
```